.github/workflows/doc-build.yml

```diff
@@ -21,10 +21,45 @@ env:
   JOB_COUNT: 4
 
 jobs:
+  doc-file-check:
+    name: Check for doc changes
+    runs-on: ubuntu-24.04
+    if: >
+      github.repository_owner == 'zephyrproject-rtos'
+    outputs:
+      file_check: ${{ steps.check-doc-files.outputs.any_modified }}
+    steps:
+    - name: checkout
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 0
+    - name: Check if Documentation related files changed
+      uses: tj-actions/changed-files@2f7c5bfce28377bc069a65ba478de0a74aa0ca32 # v46.0.1
+      id: check-doc-files
+      with:
+        files: |
+          doc/
+          boards/**/doc/
+          **.rst
+          include/
+          kernel/include/kernel_arch_interface.h
+          lib/libc/**
+          subsys/testsuite/ztest/include/**
+          **/Kconfig*
+          west.yml
+          scripts/dts/
+          doc/requirements.txt
+          .github/workflows/doc-build.yml
+          scripts/pylib/pytest-twister-harness/src/twister_harness/device/device_adapter.py
+          scripts/pylib/pytest-twister-harness/src/twister_harness/helpers/shell.py
+
   doc-build-html:
     name: "Documentation Build (HTML)"
+    needs: [doc-file-check]
     if: >
-      github.repository_owner == 'zephyrproject-rtos'
+      github.repository_owner == 'zephyrproject-rtos' &&
```

Comment:
Should this check be repeated if it already needs `doc-file-check`?

Reply:
yeah actually I've removed the gating altogether -- workflow doesn't require a specific runner and being able to have doc built in ones personal fork can actually be quite convenient